src/journal/journal-file.c

```diff
@@ -1798,41 +1798,8 @@ static int link_entry_into_array(JournalFile *f,
         return 0;
 }
 
-static int link_entry_into_array_plus_one(JournalFile *f,
-                                          le64_t *extra,
-                                          le64_t *first,
-                                          le64_t *idx,
-                                          uint64_t p) {
```

Comment:
I ma not sure i like this. I don#t think this is an improvement.

A comment might be a better way to handle this.

the "plus one" means that the array this links into has a separate "plus one" entry in the referencing structure. i.e. when we link up an array with the originating object the object typically stores hte first array entry in the originating object itself, and only the second and all others in the array.

See: https://www.freedesktop.org/wiki/Software/systemd/journal-files/

The part about "Data Objects", look for the "plus 1" in there, and what it refers to.

I kinda like the abstraction that was in place here so far: i.e. we have one helper that adds an entry to the chained up array, and than a wrapper around it that adds support for the special handling of the first entry (the "plus one").

I am not convinced this change is really an improvement.

Reply:
I found out what "plus one" was about but "+1" didn't suggest the special handling of the first entry to me. But more importantly and as the commit message explained this function has the bad side effect to obfuscate the parameters passed from link_entry_into_array() for no real gain. But maybe it's just me...

I'm just find to drop the cleanup stuff from this series the most important part is only the fix regarding the logging of journald itself anyway.